OpenRA.Game/Widgets/WidgetLoader.cs

```diff
@@ -48,6 +48,15 @@ public Widget LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node)
 			if (!args.ContainsKey("modRules"))
 				args = new WidgetArgs(args) { { "modRules", modData.DefaultRules } };
 
+			object parentLogic = null;
+			if (args.ContainsKey("logic"))
+			{
+				parentLogic = args["logic"];
+				args.Remove("logic");
+			}
+
+			args.Add("logic", node.Value.Nodes.FirstOrDefault(n => n.Key == "Logic"));
```

Comment:
I think it would be a lot nice to ToDictionary the child nodes, and then store it under a `LogicArgs` key.  You then don't need to stash and restore `Logic`, and the code that uses these becomes simpler.